src/Simplify_Max.cpp

```diff
@@ -69,10 +81,10 @@ Expr Simplify::visit(const Max *op, ExprInfo *bounds) {
              rewrite(max(min(y, x), x), b) ||
              rewrite(max(min(x, c0), c1), b, c1 >= c0) ||
 
-             rewrite(max(intrin(Call::likely, x), x), a) ||
-             rewrite(max(x, intrin(Call::likely, x)), b) ||
-             rewrite(max(intrin(Call::likely_if_innermost, x), x), a) ||
-             rewrite(max(x, intrin(Call::likely_if_innermost, x)), b) ||
+             rewrite(max(intrin(Call::likely, x), x), b) ||
```

Comment:
Is the reordering here significant?

Reply:
It's not reordering, it's changing it to return the other arg. So max(likely(x), x) now returns x instead of likely(x)